src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs

```diff
@@ -2606,6 +2811,52 @@ internal static extern
                                     ref JOBOBJECT_BASIC_PROCESS_ID_LIST lpJobObjectInfo,
                                     int cbJobObjectLength, IntPtr lpReturnLength);
 
+        internal static bool QueryServiceConfig(NakedWin32Handle hService, out IntPtr structurePointer)
+        {
+            IntPtr lpBuffer = IntPtr.Zero;
+            DWORD bufferSize, bufferSizeNeeded = 0;
+            bool status = NativeMethods.QueryServiceConfigW(
+                hSCManager: hService,
+                lpServiceConfig: lpBuffer,
+                cbBufSize: 0,
+                pcbBytesNeeded: out bufferSizeNeeded);
+
+            lpBuffer = Marshal.AllocCoTaskMem((int)bufferSizeNeeded);
+            bufferSize = bufferSizeNeeded;
+
+            status = NativeMethods.QueryServiceConfigW(
+                hService,
+                lpBuffer,
+                bufferSize,
+                out bufferSizeNeeded);
+            structurePointer = lpBuffer;
+            return status;
+        }
+
+        internal static bool QueryServiceConfig2(NakedWin32Handle hService, DWORD infolevel, out IntPtr structurePointer)
+        {
+            IntPtr lpBuffer = IntPtr.Zero;
+            DWORD bufferSize, bufferSizeNeeded = 0;
+            bool status = NativeMethods.QueryServiceConfig2W(
+                hService: hService,
+                dwInfoLevel: infolevel,
+                lpBuffer: lpBuffer,
+                cbBufSize: 0,
+                pcbBytesNeeded: out bufferSizeNeeded);
```

Comment:
we should return immediately if status != true && GetLastError() != ERROR_INSUFFICIENT_BUFFER

Reply:
resolved